hazelcast/src/main/java/com/hazelcast/quorum/impl/ProbabilisticQuorumFunction.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum.impl;
+
+import com.hazelcast.core.Member;
+import com.hazelcast.core.MemberAttributeEvent;
+import com.hazelcast.core.MembershipEvent;
+import com.hazelcast.core.MembershipListener;
+import com.hazelcast.internal.cluster.fd.PhiAccrualClusterFailureDetector;
+import com.hazelcast.internal.cluster.fd.PingFailureDetector;
+import com.hazelcast.quorum.HeartbeatAware;
+import com.hazelcast.quorum.PingAware;
+import com.hazelcast.quorum.QuorumFunction;
+import com.hazelcast.spi.properties.HazelcastProperties;
+import com.hazelcast.util.Clock;
+
+import java.util.Collection;
+import java.util.Properties;
+
+import static com.hazelcast.internal.cluster.fd.PhiAccrualClusterFailureDetector
+        .HEARTBEAT_PHI_FAILURE_DETECTOR_MIN_STD_DEV_MILLIS;
+import static com.hazelcast.internal.cluster.fd.PhiAccrualClusterFailureDetector.HEARTBEAT_PHI_FAILURE_DETECTOR_SAMPLE_SIZE;
+import static com.hazelcast.internal.cluster.fd.PhiAccrualClusterFailureDetector.HEARTBEAT_PHI_FAILURE_DETECTOR_THRESHOLD;
+
+public class ProbabilisticQuorumFunction
+        implements HeartbeatAware, QuorumFunction, MembershipListener, PingAware {
+
+    private final double phiThreshold;
+    private final int quorumSize;
+    private final int maxSampleSize;
+    private final long minStdDeviationMillis;
+    private final long acceptableHeartbeatPauseMillis;
+    private final long heartbeatIntervalMillis;
+    private final PhiAccrualClusterFailureDetector failureDetector;
+    private PingFailureDetector pingFailureDetector;
+
+    public ProbabilisticQuorumFunction(int quorumSize, long heartbeatIntervalMillis,
+                                       long acceptableHeartbeatPauseMillis,
+                                       int maxSampleSize,
+                                       long minStdDeviationMillis,
+                                       double phiThreshold) {
+        this.heartbeatIntervalMillis = heartbeatIntervalMillis;
+        this.acceptableHeartbeatPauseMillis = acceptableHeartbeatPauseMillis;
+        this.maxSampleSize = maxSampleSize;
+        this.minStdDeviationMillis = minStdDeviationMillis;
+        this.phiThreshold = phiThreshold;
+        this.quorumSize = quorumSize;
+
+        HazelcastProperties parameters = parametersAsProperties();
+        this.failureDetector = new PhiAccrualClusterFailureDetector(acceptableHeartbeatPauseMillis,
```

Comment:
What about adding a new constructor to `PhiAccrualClusterFailureDetector`, instead of creating `HazelcastProperties` from parameters and parsing parameters again?